lib/Runtime/Library/JavascriptLibrary.cpp

```diff
@@ -3873,10 +3876,14 @@ namespace Js
         JavascriptFunction ** builtinFuncs = library->GetBuiltinFunctions();
 
         library->AddMember(regexPrototype, PropertyIds::constructor, library->regexConstructor);
+        library->regexConstructorSlotIndex = 0;
+        Assert(regexPrototype->GetSlot(library->regexConstructorSlotIndex) == library->regexConstructor);
```

Comment:
:smile: I agree that the code doesn't look good with the hard coded values and would require updates if a new property is added in between. It doesn't seem like it would be quick to add such an API, so I think hard coding is enough for now. I'll get back to this later.